### DISCUSSION TITLE
Tine20 auf Plesk (Strato)

### DISCUSSION THREAD

**Discussion by Patriot2k** [tine20/tine20#7437] [General]
Schönen guten Abend!

Ein ehrenamtlicher Verein hat mich gebeten ihre aktuelle Tine20 Installation (2020.03.02) auf die aktuellste Version zu bringen.
Gehostet wird das Ganze bei Strato auf einer Ubuntu 18.04 LTS. 
Die aktuelle PHP Version ist leider < 7.4, daher ist ein einfaches Update nicht möglich.
Zusätzlich ist die 18.04 nicht mehr im Support und ein Update macht an der Stelle auch Sinn.

Meine Überlegung ist daher, das ich direkt auf 22.04 update und Plesk zur Verwaltung nutze.
Wenn ich allerdings die OS Version bei Strato ändere, verliere ich auch Zugriff auf die automatischen Backups.
Auf Grund der Kritikalität der gespeicherten Daten bin ich gerade dabei mir eine VM mit einer funktionierenden Instanz aufzusetzen.

Ich stehe jetzt vor der Herausforderung das ich bei Plesk eine Installation in dem Ordner: 
/var/www/vhosts/XXXXXXXX.plesk.page/httpdocs/tine/ durchführen möchte, die Tine Installation allerdings /var/www/tine vorsieht.
Hat da schon jemand Erfahrung mit, bzw. ist das überhaupt möglich.
Mein Wunsch wäre es das ich das Ganze nach dem Update einfach administrierbar an jemand anders übergebe.

Aktueller Stand: 
VMWare VM mit Ubuntu 22.04 und ich bin dabei Tine 2020.03.02 zu installieren um den Migrationspfad testen zu können.
Backups von der originalen SQL Tine20 DB habe ich natürlich angefertigt.

Vielleicht könnt ihr mich ja "in die richtige Richtung" schubsen.

Vielen lieben Dank im Voraus,
Robert

**Comment by Patriot2k** [comment]
Wäre es vielleicht sinnvoll Tine per Docker zu installieren?

**Reply by lab-at-nohl** [reply]
Ich habe mit Docker kaum Erfahrung. Aber ja, das ist der Weg den die Tine-Macher gehen wollen. Mit längerfristigem Support auch der Community darf man also wohl rechnen... Tatsächlich hilft dir das bei dem Problem mit der php-Version. Und du bekommst "einfach" Updates. 

Auf deb-Pakete würde ich mich nicht verlassen. Ich weiß noch nicht mal, ob es welche für 22.04 gibt. Jedenfalls werden spätere Updates sicherlich nicht als Distro packages für dann ggf. veraltete Distros verfügbar sein. 

Ich persönlich setze nicht auf Docker, sondern auf die ZIP files, dafür habe ich ein Update-Script, um nicht jedes Mal von vorne anzufangen. Das bringt die notwendige Flexibilität zwischen Tine-Version und Distro. 

Neben dem Docker Image brauchst du auf jeden Fall MariaDB und IMAP/Postfix. Da die sehr zeitlos sind, kann man diese auch direkt auf dem Server installieren. Wenn du also bei 18.04 (eol?) bleiben willst, ist Docker definitiv ein guter Weg. Wenn du alles neu machst, kann man das machen, muss aber nicht.

**Comment by lab-at-nohl** [comment]
Offenbar musst du in 22.04 wohl mittels Plesk ein System aufsetzen, das alles aus dem 18.04 Server mitbringt. Datenbank kann man sicher einfach kopieren. Wie sieht es mit den Mails aus? Wie ist der alte Server gestrickt, authentifizieren IMAP/Postfix gegen Tine mittels MySQL oder wie funktioniert Mail bei dir? Grundsätzlich brauchst du erstmal einen funktionierenden Server... Ich fürchte, das wird man zwar alles mit Plesk irgendwie machen können, aber nicht mal eben so...

Ein fest voreingestelltes Tine-Verzeichnis gibt es nur, wenn du deb Pakete verwendest. Deshalb nimm die ZIPs. Was dann noch fehlt ist der Worker, in dem deb sind nur sehr wenige Dateien, die kann man entpacken und manuell anpassen. 

Versuche keinesfalls zugleich ein Upgrade, sondern erstmal nur den Umzug wuppen. Und wenn alles läuft, dann Tine upgraden...

**Comment by Patriot2k** [comment]
Vielen Dank für deine Rückmeldung!
Die aktuelle Tine Installation wird nur für die Kalender Funktion und die Dateiablage genutzt.
Mail kann ich also aktuell ignorieren.

Wenn ich dich richtig verstehe schlägst du vor das ich auf einem neuen Ubuntu Server die alte EOL Tine Version installiere, die DB importiere und zum laufen bekomme und anschließend die Tine Version über die bekannten Migrationspfade upgrade.
--> Sollte machbar sein.

Wie kompatible sind denn die Tine20 DB Stände? Kann ich aus einer alten Tine exportieren und in einer neuen Version importieren?
Würde mich wundern, aber die Hoffnung stirbt zuletzt :)

**Comment by lab-at-nohl** [comment]
Ich wusste nicht, dass Tine ganz ohne Mail geht. Aber warum nicht. 

Das `filesdir` musst du natürlich auch kopieren. Und ja, so ist mein Vorschlag. Du kannst ja aus /var/www/tine die Dateien kopieren und dann die `config.inc.php` anpassen, falls notwendig. Und schau Mal, ob du in /etc/tine20 Dateien hast. Schließlich würde ich mit `systemctl status tine20-worker` schauen, ob ein Worker läuft. 

Ein Tine-Upgrade besteht aus geänderten Programmdateien und der durch setup.php geänderten DB-Struktur. Ich würde ungern während einem Umzug zugleich das Update machen. Wenn etwas nicht geht, weißt du nicht ob es am System oder an Tine liegt. Wenn man deb Pakete nutzt geht es allerdings möglicherweise nicht anders...

**Comment by Patriot2k** [comment]
Moin,
mittlerweile bin ich schon ein ganz schönes Stück weitergekommen:
Ich habe eine lokale VM mit Tine 2020.03.02 und Ubuntu 22 aufgebaut und den folgenden Migrationspfad durchlaufen:
2020.03.02 -> 2020.03.4 -> 2020.08.8 -> 2021.02.4 -> 2021.12.1
Stolpersteine waren bis jetzt das PHP8.1 nicht mit Zend funktioniert hat und ein Fallback auf PHP 7.4 notwendig war.
Bei 2021.02.4 konnte das Addressbuch nicht aktualisiert werden, hat dann aber mit 2021.12.1 funktioniert.

Bei 2021.12.1 bekomme ich bei der Anmeldung den Hinweis das MFA aktiviert werden soll, allerdings kann die Einrichtung nicht durchgeführt werden. Es gibt ein Progress Bar mit dem Hinweis das die Anmeldung durchgeführt wird, aber kann nicht abgeschlossen werden.

Im Tine20.log sehe ich das Folgende:
Zu dem Failed to create hash file habe ich leider nichts verwertbares finden können.
Vielleicht habt ihr ja einen Tip für mich?

3664d0 - setupuser - 2023-02-12T11:14:29+00:00 ERR (3): Setup_Controller::updateApplications::562 failed to create hash file
3664d0 - setupuser - 2023-02-12T11:14:29+00:00 ERR (3): Setup_Controller::updateApplications::563 #0 /var/www/tine/Tinebase/FileSystem.php(643): Tinebase_FileSystem->createFileBlob()
#1 /var/www/tine/Setup/Controller.php(2109): Tinebase_FileSystem->fclose()
#2 /var/www/tine/Setup/Controller.php(2134): Setup_Controller->{closure}()
#3 /var/www/tine/Setup/Controller.php(584): Setup_Controller->createImportExportDefinitions()
#4 /var/www/tine/Setup/Controller.php(556): Setup_Controller->updateAllImportExportDefinitions()
#5 /var/www/tine/Setup/Frontend/Json.php(118): Setup_Controller->updateApplications()
#6 /var/www/tine/vendor/zendframework/zendframework1/library/Zend/Server/Abstract.php(231): Setup_Frontend_Json->updateApplications()
#7 /var/www/tine/vendor/zendframework/zendframework1/library/Zend/Json/Server.php(603): Zend_Server_Abstract->_dispatch()
#8 /var/www/tine/vendor/zendframework/zendframework1/library/Zend/Json/Server.php(206): Zend_Json_Server->_handle()
#9 /var/www/tine/Setup/Server/Json.php(90): Zend_Json_Server->handle()
#10 /var/www/tine/Setup/Core.php(117): Setup_Server_Json->handle()
#11 /var/www/tine/setup.php(14): Setup_Core::dispatchRequest()
#12 {main}
3664d0 - setupuser - 2023-02-12T11:14:29+00:00 ERR (3): Tinebase_Server_Json::_handleException::434 Tinebase_Exception_UnexpectedValue -> failed to create hash file
3664d0 - setupuser - 2023-02-12T11:14:29+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::133 Tinebase_Exception_UnexpectedValue -> failed to create hash file
3664d0 - setupuser - 2023-02-12T11:14:29+00:00 ERR (3): Tinebase_Exception::logExceptionToLogger::155 #0 .../Tinebase/FileSystem.php(643): Tinebase_FileSystem->createFileBlob()
#1 .../Setup/Controller.php(2109): Tinebase_FileSystem->fclose()
#2 .../Setup/Controller.php(2134): Setup_Controller->{closure}()
#3 .../Setup/Controller.php(584): Setup_Controller->createImportExportDefinitions()
#4 .../Setup/Controller.php(556): Setup_Controller->updateAllImportExportDefinitions()
#5 .../Setup/Frontend/Json.php(118): Setup_Controller->updateApplications()
#6 .../vendor/zendframework/zendframework1/library/Zend/Server/Abstract.php(231): Setup_Frontend_Json->updateApplications()
#7 .../vendor/zendframework/zendframework1/library/Zend/Json/Server.php(603): Zend_Server_Abstract->_dispatch()
#8 .../vendor/zendframework/zendframework1/library/Zend/Json/Server.php(206): Zend_Json_Server->_handle()
#9 .../Setup/Server/Json.php(90): Zend_Json_Server->handle()
#10 .../Setup/Core.php(117): Setup_Server_Json->handle()
#11 .../setup.php(14): Setup_Core::dispatchRequest()
#12 {main}

**Reply by lab-at-nohl** [reply]
Das klingt doch alles erst mal sehr gut, du bist dort angelangt wo du sein wolltest. Dir stellt sich jetzt also die Frage, ob du es dabei belässt bis zum nächsten jährlichen Upgrade...

Der beschriebene Fehler wurde m.E. später gefixt, allerdings in den wöchentlichen Nightlies. Das einfachste ist es daher, die MFA zu deaktivieren (ich glaube, das war in Admin -> Rollen -> Tinebase). Oder, ob du spielst ein Nightly ein, in dem der Fix enthalten ist.

**Reply by Patriot2k** [reply]
Aktuell bin ich auf der Version 2021.12.1. Um wirklich aktuell zu sein fehlt doch noch der Sprung auf 2022.12.1 oder etwa nicht?
Sobald ich wieder zuhause bin probiere ich es mal mit einer deaktivierten MFA.

Vielen Dank für deine Rückmeldung!

**Reply by lab-at-nohl** [reply]
Ups, ja stimmt aktuell ist 20**22**.12.1. Hatte ich zu schnell gelesen. Ich hatte erst dort die MFA-Probleme. Im Übrigen ist der Log-Auszug ja ein Update-Fehler - nicht Login. Mithin muss zunächst das Update erfolgreich durchlaufen. Ich konnte nur mit dem Log-Auszug wenig anfangen, das scheint nicht level 7 sondern 5 zu sein (der Fehler könnte in jeder app sein, da der Weg dorthin nicht gezeigt wird)... In der config.inc.php ist das `'filesdir' => ...` vorhanden hoffe ich. 

Bitte poste nochmal ein Log mit level `DEBUG` = 7 vom Update (einfach nochmal laufen lassen). Du hast doch eine Konsole, oder? Dann lass das Update dort laufen, dadurch kann man das passende Stück Log leichter finden, weil nur genau das Update und sonst nix passiert (z.B. einfach die Log-Datei vorher leeren). Update über Konsole geht mit `php setup.php --update`, bewirkt dasselbe wie im wenn man setup.php über den Browser aufruft.
